hibernate-core/src/main/java/org/hibernate/sql/results/internal/StandardRowReader.java

```diff
@@ -102,6 +114,10 @@ public T readRow(RowProcessingState rowProcessingState, JdbcValuesSourceProcessi
 		return rowTransformer.transformRow( resultRow );
 	}
 
+	private <k> k[] getResultRowArray(Class<k> clazz, int size) {
+		return (k[]) Array.newInstance( clazz, size );
```

Comment:
This requires reflective access, which is a problem for e.g. Quarkus. I guess the JVM is probably also not very happy with this sort of code, but it's just speculation.
Either way, IMO we should rather handle this by using a special `RowTransformer`.

Reply:
It feels like this change could have a perf impact, whereas pushing that work into a special `RowTransformer` would not be a big deal probably, but not be on the "hot path".